src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/storage/OcrValidationRetryManager.java

```diff
@@ -0,0 +1,121 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.storage;
+
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static java.time.LocalDateTime.now;
+import static java.time.LocalDateTime.parse;
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.bulkscanprocessor.util.TimeZones.EUROPE_LONDON_ZONE_ID;
+
+@Component
+public class OcrValidationRetryManager {
+
+    private static final Logger logger = getLogger(OcrValidationRetryManager.class);
+
+    private static final String RETRY_COUNT_METADATA_PROPERTY = "ocrValidationRetryCount";
+    private static final String RETRY_DELAY_EXPIRATION_TIME_METADATA_PROPERTY =
+        "ocrValidationRetryDelayExpirationTime";
+
+    private final LeaseAcquirer leaseAcquirer;
+    private final int ocrValidationMaxRetries;
+    private final int ocrValidationRetryDelaySec;
+
+    public OcrValidationRetryManager(
+        LeaseAcquirer leaseAcquirer,
+        @Value("${ocr-validation-max-retries}") int ocrValidationMaxRetries,
+        @Value("${ocr-validation-delay-retry-sec}") int ocrValidationRetryDelaySec
+    ) {
+        this.leaseAcquirer = leaseAcquirer;
+        this.ocrValidationMaxRetries = ocrValidationMaxRetries;
+        this.ocrValidationRetryDelaySec = ocrValidationRetryDelaySec;
+    }
+
+    public boolean isReadyToRetry(BlobClient blobClient) {
+        Map<String, String> blobMetaData = blobClient.getProperties().getMetadata();
+        String retryDelayExpirationTime = blobMetaData.get(RETRY_DELAY_EXPIRATION_TIME_METADATA_PROPERTY);
+
+        final boolean retryDelayExpired = isRetryDelayExpired(retryDelayExpirationTime);
+
+        if (!retryDelayExpired) {
+            logger.info(
+                "Retry delay not yet expired on file {} in container {}",
+                blobClient.getBlobName(),
+                blobClient.getContainerName()
+            );
+        }
+
+        return retryDelayExpired;
+    }
+
+    /**
+     * Check if number of retries is exceeding ocrValidationMaxRetries,
+     * if not set new retry delay expiration time, increment retryCount and return true,
+     * otherwise return false which means no more retries are possible.
+     *
+     * @return true if retry is possible and new retry delay expiration time has been set,
+     *         false if no more retries are possible
+     */
+    public boolean setRetryDelayIfPossible(BlobClient blobClient) {
+        Map<String, String> blobMetaData = blobClient.getProperties().getMetadata();
```

Comment:
this should be done after blob leased. 
Meantime another tread can increase the number but thread would continue with old value.

Reply:
Agree with you here